nixos/modules/programs/wayland/hyprland.nix

```diff
@@ -85,6 +96,22 @@ in
         };
       }
 
+      (lib.mkIf (cfg.withUWSM) {
+        programs.uwsm.enable = true;
+        # Configure UWSM to launch Hyprland from a display manager like SDDM
+        programs.uwsm.waylandCompositors = {
+          hyprland = {
+            prettyName = "Hyprland";
+            comment = "Hyprland compositor managed by UWSM";
+            binPath = "/run/current-system/sw/bin/Hyprland";
```

Comment:
I'm a bit late but I don't think this is what we want. We want 

```Suggestion
            binPath = lib.getExe config.programs.hyprland.package;
```

because we want the desktop-manager to be built from the same revision as the compositor. For instance, GDM crashes if there's a mesa version mismatch and the same probably goes for a glibc mismatch.

The problem here is that `desktop-manager.service` does not restart upon NixOS activation (as to not kill your session) and is therefore constant. The global bin is a moving target however which makes a mismatch possible.

(The desktop sessions (sessionPackages) are hard-coded into the `display-manager.service` via env var which only changes when the service is restarted rather than from a systemd-wide share directory, so store paths also remain constant.)

Reply:
> if two packages providing the same `bin` at the same time, it will conflict AFAIK during rebuild and fail.

That has nothing to do with the issue I described. We're not adding any Hyprland binaries to the global environment here.

> This actually isn't an issue since Hyprland wayland-sessions desktop file itself just has `Exec=Hyprland`

The desktop file is not used. You explicitly set uwsm to use the binary from the global bin path here. Even if it used the desktop file it'd have the exact same issue.